From: Jiri Slaby <jslaby@suse.cz>
Subject: kABI: reintroduce vlan_untag
Patch-mainline: never, kabi

In 3.12.31, commit f55ef858b1beb467609805086a18a0e91857e577
(net: Always untag vlan-tagged traffic on input.), upstream commit
0d5501c1c828fb97d02af50aa9d2b1a5498b94e4 renamed the exported function
vlan_untag to skb_vlan_untag and moved it to net/core/skbuff.c.
Therefore, it makes the symbol invisible to kABI checker and potential
users.

Reintroduce the function, just as a wrapper for skb_vlan_untag.

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 include/linux/if_vlan.h |    1 +
 net/8021q/vlan_core.c   |    6 ++++++
 2 files changed, 7 insertions(+)

--- a/include/linux/if_vlan.h
+++ b/include/linux/if_vlan.h
@@ -90,6 +90,7 @@ extern struct net_device *vlan_dev_real_
 extern u16 vlan_dev_vlan_id(const struct net_device *dev);
 
 extern bool vlan_do_receive(struct sk_buff **skb);
+extern struct sk_buff *vlan_untag(struct sk_buff *skb);
 
 extern int vlan_vid_add(struct net_device *dev, __be16 proto, u16 vid);
 extern void vlan_vid_del(struct net_device *dev, __be16 proto, u16 vid);
--- a/net/8021q/vlan_core.c
+++ b/net/8021q/vlan_core.c
@@ -106,6 +106,12 @@ u16 vlan_dev_vlan_id(const struct net_de
 }
 EXPORT_SYMBOL(vlan_dev_vlan_id);
 
+struct sk_buff *vlan_untag(struct sk_buff *skb)
+{
+	return skb_vlan_untag(skb);
+}
+EXPORT_SYMBOL(vlan_untag);
+
 /*
  * vlan info and vid list
  */
